parl/remote/remote_decorator.py

```diff
@@ -116,26 +103,31 @@ def _connect_server(self, server_ip, server_port, remote_ip,
             socket = self.zmq_context.socket(zmq.REQ)
             socket.connect("tcp://{}:{}".format(server_ip, server_port))
 
-            client_id = np.random.randint(int(1e18))
-            logger.info("client_id:{}".format(client_id))
             logger.info("connecting {}:{}".format(server_ip, server_port))
 
-            client_info = '{}:{} {}'.format(local_ip, local_port, client_id)
+            client_addr = '{}:{}'.format(local_ip, local_port)
             socket.send_multipart(
                 [remote_constants.CONNECT_TAG,
-                 to_byte(client_info)])
+                 to_byte(client_addr)])
 
             message = socket.recv_multipart()
-            logger.info("[connect_server] done, message from server:{}".format(
-                message))
+            self.client_id = message[1]
+            logger.info("connect server done, client_id: {}".format(
+                self.client_id))
             self.connect_socket = socket
             self.connect_socket.linger = 0
 
         def _exit_remote(self):
-            # Following release order matters
             self.poller.unregister(self.connect_socket)
 
-            self.zmq_context.destroy()
+            self.connect_socket.close()
+            self.reply_socket.close()
+
+            # The program may hang when destroying zmq context manually.
+            # So we will let it destroyed by garbage collection of python, which may raise
```

Comment:
So we will let it destroyed by the garbage collection mechanism of python, which may raise

Reply:
done